hazelcast/src/test/java/com/hazelcast/config/MapConfigTest.java

```diff
@@ -312,8 +312,8 @@ public void givenCacheDeserializedValuesSetToALWAYS_whenSetOptimizeQueriesToFals
         mapConfig.setOptimizeQueries(false);
     }
 
-    @Test(expected = ConfigurationException.class)
-    public void givenCacheDeserializedValuesSetToINDEX_ONLY_whenSetOptimizeQueriesToFalse_thenThrowConfigurationException() {
+    @Test
+    public void givenCacheDeserializedValuesSetToINDEX_ONLY_whenSetOptimizeQueriesToFalse_thenNoException() {
```

Comment:
At the risk of becoming really annoying now: what about adding `givenCacheDeserializedValuesSetToNEVER_whenSetOptimizeQueriesToFalse`, too?